pkgs/development/interpreters/python/cpython/default.nix

```diff
@@ -260,7 +260,7 @@ in with passthru; stdenv.mkDerivation {
   ] ++ optionals (pythonAtLeast "3.9" && pythonOlder "3.11" && stdenv.isDarwin) [
     # Stop checking for TCL/TK in global macOS locations
     ./3.9/darwin-tcl-tk.patch
-  ] ++ optionals (isPy3k && hasDistutilsCxxPatch) [
+  ] ++ optionals (isPy3k && hasDistutilsCxxPatch && pythonOlder "3.12") [
```

Comment:
Is now linux broken or is this unrelated?

```
building '/nix/store/ynv4z5p29s6zs84l6w6g0232zq42jiy0-python3-3.12.0a3-env.drv'...
created 252 symlinks in user environment
building '/nix/store/7vczn2znq9b9bkrv97cbsdr75zn64dr9-python3-3.12.0a3-site-prefix-mypy-test.drv'...
Traceback (most recent call last):
  File "/nix/store/f6k6kv22np8n8x71k6fbzpmnc7dlk6x4-python3.12-mypy-0.991/bin/.mypy-wrapped", line 6, in <module>
    from mypy.__main__ import console_entry
  File "/nix/store/y3xpjxh3n1apqzppr82vmabb1sv8sigj-python3-3.12.0a3-env/lib/python3.12/site-packages/mypy/__main__.py", line 9, in <module>
    from mypy.main import main, process_options
  File "mypy/main.py", line 215, in <module>
TypeError: type AugmentedHelpFormatter has the Py_TPFLAGS_MANAGED_DICT flag but tp_dictoffset is set
error: 1 dependencies of derivation '/nix/store/1xrgh2nnmxhm6sp3i5ashr9456ckbrcc-python3-3.12.0a3-env.drv' failed to build
error: builder for '/nix/store/7vczn2znq9b9bkrv97cbsdr75zn64dr9-python3-3.12.0a3-site-prefix-mypy-test.drv' failed with exit code 1;
       last 7 log lines:
       > Traceback (most recent call last):
       >   File "/nix/store/f6k6kv22np8n8x71k6fbzpmnc7dlk6x4-python3.12-mypy-0.991/bin/.mypy-wrapped", line 6, in <module>
       >     from mypy.__main__ import console_entry
       >   File "/nix/store/y3xpjxh3n1apqzppr82vmabb1sv8sigj-python3-3.12.0a3-env/lib/python3.12/site-packages/mypy/__main__.py", line 9, in <module>
       >     from mypy.main import main, process_options
       >   File "mypy/main.py", line 215, in <module>
       > TypeError: type AugmentedHelpFormatter has the Py_TPFLAGS_MANAGED_DICT flag but tp_dictoffset is set
       For full logs, run 'nix log /nix/store/7vczn2znq9b9bkrv97cbsdr75zn64dr9-python3-3.12.0a3-site-prefix-mypy-test.drv'.
error: 1 dependencies of derivation '/nix/store/k9vvfa5wfl2ha29rf2ii9i9a3a4yrs8x-python3-3.12.0a3-tests-nixenv-virtualenv.drv' failed to build
error: 1 dependencies of derivation '/nix/store/yxlxiqxprgaglk0ab7r0gwyzdlzn64nm-import-requests.drv' failed to build
error: build of '/nix/store/1xrgh2nnmxhm6sp3i5ashr9456ckbrcc-python3-3.12.0a3-env.drv', '/nix/store/7vczn2znq9b9bkrv97cbsdr75zn64dr9-python3-3.12.0a3-site-prefix-mypy-test.drv', '/nix/store/k9vvfa5wfl2ha29rf2ii9i9a3a4yrs8x-python3-3.12.0a3-tests-nixenv-virtualenv.drv', '/nix/store/yxlxiqxprgaglk0ab7r0gwyzdlzn64nm-import-requests.drv' failed
```

Reply:
`hasDistutilsCxxPatch = !(stdenv.cc.isGNU or false)` evaluates to false on Linux, so the failure cannot be due to this PR.